app/controllers/repositories_controller.rb

```diff
@@ -104,14 +108,50 @@ def add_like # MAKE A LIKE CONTROLLER TO PUT THIS IN
       render json: { failed: true }
   end
 
+  def password_entry
+  end
+
+  def pass_authenticate
+    @auth = Repository.authenticate(params[:slug], params[:password])
+    @repository = Repository.find_by_slug(params[:slug])
+    respond_to do |format|
+      if @auth
+        @authorized = true
+        authorized_repo_ids << params[:id]
+        flash[:notice] = "Success"
+        format.html{redirect_to repository_path}
```

Comment:
Does this need an id for the repository?

Reply:
fixed